OpenRA.Game/Graphics/Util.cs

```diff
@@ -20,12 +20,27 @@ public static class Util
 		// yes, our channel order is nuts.
 		static readonly int[] ChannelMasks = { 2, 1, 0, 3 };
 
-		public static void FastCreateQuad(Vertex[] vertices, in float3 o, Sprite r, int2 samplers, float paletteTextureIndex, int nv, in float3 size, in float3 tint, float alpha)
+		public static void FastCreateQuad(Vertex[] vertices, in float3 o, Sprite r, int2 samplers, float paletteTextureIndex, int nv,
+			in float3 size, in float3 tint, float alpha, in float rotation = 0F)
 		{
+			var a = o;
 			var b = new float3(o.X + size.X, o.Y, o.Z);
 			var c = new float3(o.X + size.X, o.Y + size.Y, o.Z + size.Z);
 			var d = new float3(o.X, o.Y + size.Y, o.Z + size.Z);
-			FastCreateQuad(vertices, o, b, c, d, r, samplers, paletteTextureIndex, tint, alpha, nv);
+
+			// Rotate sprite if rotation angle is not equal to 0
+			if (rotation != 0F)
+			{
+				var centerPoint = o + (size / 2);
+				var angleSin = (float)Math.Sin(-rotation);
+				var angleCos = (float)Math.Cos(-rotation);
```

Comment:
`Math.SinCos`? Hopefully Mono has it.

Reply:
What is the status here? Does Mono not have `Math.SinCos`?

Reply:
I think it does because the Math class is showing compatibility with Xamarin 3.0 on the official Microsoft doc [here](https://docs.microsoft.com/en-us/dotnet/api/system.math?view=net-5.0) 
![image](https://user-images.githubusercontent.com/1495872/153021267-c9caa792-b9b5-47a5-960b-96cf1d632242.png), and it would be odd if they didn't support the Sin and Cos functions, but I will I also get this tested and report back.